examples/apps/cli/CMakeLists.txt

```diff
@@ -40,25 +40,39 @@ set(COMMON_INCLUDES
     ${PROJECT_SOURCE_DIR}/examples/platforms
     ${PROJECT_SOURCE_DIR}/src/core
 )
-
 target_include_directories(ot-cli-ftd PRIVATE ${COMMON_INCLUDES})
 target_include_directories(ot-cli-mtd PRIVATE ${COMMON_INCLUDES})
+if(OT_BUILTIN_MBEDTLS)
+    target_link_libraries(ot-cli-ftd
+        openthread-cli-ftd
+        ${OT_PLATFORM_LIB}
+        openthread-ftd
+        ${OT_PLATFORM_LIB}
+        mbedcrypto
```

Comment:
we can use [generator expression](https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html#conditional-generator-expressions) to include `mbedtls` optionally. for example:
```cmake
$<OT_BUILTIN_MBEDTLS:mbedcrypto>
```
so that we can remove the outer `if()...else()...` block.